pkg/mark/attachment.go

```diff
@@ -159,14 +155,51 @@ func ResolveAttachments(
 	return attaches, nil
 }
 
-func prepareAttachments(base string, replacements []string) ([]Attachment, error) {
+func ResolveLocalAttachments(opener vfs.Opener, base string, replacements []string) ([]Attachment, error) {
+	attaches, err := prepareAttachments(opener, base, replacements)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, attach := range attaches {
+		checksum, err := GetChecksum(bytes.NewReader(attach.FileBytes))
+		if err != nil {
+			return nil, karma.Format(
+				err,
+				"unable to get checksum for attachment: %q", attach.Name,
+			)
+		}
+
+		attach.Checksum = checksum
+	}
+	return attaches, err
+}
+
+func prepareAttachments(opener vfs.Opener, base string, replacements []string) ([]Attachment, error) {
 	attaches := []Attachment{}
 	for _, name := range replacements {
+		attachmentPath := filepath.Join(base, name)
```

Comment:
this func grew up a bit, worth doing something like this:
```
func prepareAttachments(opener vfs.Opener, base string, replacements []string) ([]Attachment, error) {
    attachments := make([]Attachment, 0, len(replacements))

    for _, name := range replacements {
        attachment, err := createAttachment(opener, base, name)
        if err != nil {
            return nil, err
        }

        attachments = append(attachments, attachment)
    }

    return attachments, nil
}

func createAttachment(opener vfs.Opener, base, name string) (Attachment, error) {
    attachmentPath := filepath.Join(base, name)
    file, err := opener.Open(attachmentPath)
    if err != nil {
        return Attachment{}, karma.Format(err, "unable to open file: %q", attachmentPath)
    }
    defer file.Close()

    fileBytes, err := io.ReadAll(file)
    if err != nil {
        return Attachment{}, karma.Format(err, "unable to read file: %q", attachmentPath)
    }

    return Attachment{
        Name:      name,
        Filename:  strings.ReplaceAll(name, "/", "_"),
        FileBytes: fileBytes,
        Replace:   name,
    }, nil
}
```

Reply:
I added a resolveAttachement function.